sympy/core/tests/test_equation.py

```diff
@@ -0,0 +1,122 @@
+
+
+def test_convert_equation():
+    a, b, c = var('a b c')
+    tsteqn = Eqn(a, b / c)
+    assert tsteqn.as_Boolean() == Equality(a, b / c)
+    assert tsteqn.reversed == Equation(b / c, a)
+    assert tsteqn.swap == Equation(b / c, a)
+
+
+def test_binary_op():
+    a, b, c = var('a b c')
+    tsteqn = Eqn(a, b / c)
+    assert tsteqn + c == Equation(a + c, b / c + c)
+    assert c + tsteqn == Equation(c + a, c + b / c)
+    assert tsteqn * c == Equation(a * c, b)
+    assert c * tsteqn == Equation(c * a, b)
+    assert tsteqn - c == Equation(a - c, b / c - c)
+    assert c - tsteqn == Equation(c - a, c - b / c)
+    assert tsteqn / c == Equation(a / c, b / c ** 2)
+    assert c / tsteqn == Equation(c / a, c ** 2 / b)
+    assert tsteqn % c == Equation(a % c, (b / c) % c)
+    assert c % tsteqn == Equation(c % a, c % (b / c))
+    assert tsteqn ** c == Equation(a ** c, (b / c) ** c)
+    assert c ** tsteqn == Equation(c ** a, c ** (b / c))
+    assert tsteqn + tsteqn == Equation(2 * a, 2 * b / c)
+    assert tsteqn * tsteqn == Equation(a ** 2, b ** 2 / c ** 2)
+    assert tsteqn - tsteqn == Equation(0, 0)
+    assert tsteqn / tsteqn == Equation(1, 1)
+    assert tsteqn % tsteqn == Equation(0, 0)
+    assert tsteqn ** tsteqn == Equation(a ** a, (b / c) ** (b / c))
+
+
+def test_outputs():
+    a, b, c = var('a b c')
+    tsteqn = Eqn(a, b / c)
+    assert str(tsteqn) == 'a = b/c'
+    assert latex(tsteqn) == 'a = \\frac{b}{c}'
+
+
+def test_helper_functions():
+    a, b, c = var('a b c')
+    tsteqn = Eqn(a, b / c)
+    raises(ValueError, lambda: integrate(tsteqn, c))
+    raises(AttributeError, lambda: integrate(tsteqn, c, side='right'))
+    assert tsteqn.evalf(4, {b: 2.0, c: 4}) == Eqn(a, 0.5000)
+    assert diff(tsteqn, c) == Equation(diff(a, c, evaluate=False), -b / c ** 2)
+    tsteqn = Eqn(a * c, b / c)
+    assert diff(tsteqn, c) == Equation(a, -b / c ** 2)
+    assert integrate(tsteqn, c, side='rhs') == integrate(tsteqn.rhs, c)
+    assert integrate(tsteqn, c, side='lhs') == integrate(tsteqn.lhs, c)
+
+    def adsq(expr):
+        # Arbitrary python function
+        return expr + expr ** 2
+
+    assert (adsq(tsteqn) == Eqn(a ** 2 * c ** 2 + a * c, b ** 2 / c ** 2 +
+                                b / c))
+    tsteqn = Eqn((a - 1) * (a + 1) * (a + 3), (a + 3) * (2 * b + c) ** 2)
+    tsteqn2 = Eqn(a ** 3 + 3 * a ** 2 - a - 3,
+                  4 * a * b ** 2 + 4 * a * b * c + a * c ** 2 + 12 * b ** 2 +
+                  12 * b * c + 3 * c ** 2)
+    tsteqn3 = Eqn(a ** 3 + 3 * a ** 2 - a - 3,
+                  a * c ** 2 + b ** 2 * (4 * a + 12) + b * (
+                      4 * a * c + 12 * c) + 3 * c ** 2)
+    tsteqn4 = Eqn(a ** 3 + 3 * a ** 2 - a - 3,
+                  a * c ** 2 + 4 * b ** 2 * (a + 3) + 4 * b * c * (
+                      a + 3) + 3 * c ** 2)
+    assert tsteqn.expand() == tsteqn2
+    assert expand(tsteqn) == tsteqn2
+    assert tsteqn2.collect(b) == tsteqn3
+    assert collect(tsteqn2, b) == tsteqn3
```

Comment:
Reading tests like this that refer to variables defined elsewhere is difficult. Can the equations just be shown explicitly in each of these tests?

(Simpler examples can be chosen to demonstrate the effect of these operations)

Reply:
Made up some simpler equations and embedded them in the assert statements.
Commit 4dc62ee7